### ISSUE TITLE
Feature request: add ByteSize() to Output class

### ISSUE THREAD

**Issue by philipp-schmidt** [NVIDIA/tensorrt-inference-server#562]
The following works quite nicely for the `Input` class:

```c++
for (const auto &input : inference_ctx->Inputs()) {
    // Calculate total byte size
    auto total_input_size = input->ByteSize() * batch_size;
}
```

The output class however has way less features and calculating the total input byte size has to be done via the `Dims` and (if I would not be lazy) `DType`:

```c++
for (const auto &output : inference_ctx->Outputs()) {
        // Calculate size of dimensions
        int dimension_size = 1;
        for (const auto &dim : output->Dims()) {
            dimension_size *= dim;
        }

        // Calculate total byte size
        size_t total_output_size = dimension_size * batch_size * sizeof(float);
}
```

Is there a reason the `Output` class has less fields?

**Comment by philipp-schmidt** [comment]
Before shared memory outputs were a thing, it was not necessary to calculate the total byte size and one could always receive it during result processing via `getRaw`. With shared memory outputs it would be a convenient feature to receive the byte size from the inference context, so registering and reserving memory is easier to maintain.

**Comment by CoderHam** [comment]
Sure, I could work on adding this feature. However, it is important to know the size of the output in the case of shared memory before making the request (or allocate more than enough space for it in the shared memory region used for the corresponding output while making the request)

**Comment by CoderHam** [comment]
Closed with #665
